File: Java #5/Checkpoint5/src/Cadastro_de_Pedidos/Programa.java
package Cadastro_de_Pedidos;

import java.util.Scanner;
import java.io.*;

public class Programa {

    public static void main(String[] args) {
        Scanner read = new Scanner(System.in);
        
        String path = "C:\\Users\\Vinicius Rodrigues\\aqq\\Desktop\\ArquivosJ\\pedidos.txt"; 
        String segundoPath = "C:\\Users\\Vinicius Rodrigues\\aqq\\Desktop\\ArquivosJ\\total_pedidos.txt"; 
        
        boolean continuar = true;
        int quantidade;
        double totalPedido = 0;
        String linha, resposta, nome = "";

        try (BufferedWriter tw = new BufferedWriter(new FileWriter(segundoPath))) {
           
        	while (continuar) {
        		
                try (BufferedReader pr = new BufferedReader(new FileReader(path))) {
                    
                    System.out.print("Digite o seu nome: ");
                    nome = read.next();

                    System.out.print("Qual a quantidade de produtos que você deseja cadastrar: ");
                    quantidade = read.nextInt();

                    for (int i = 0; i < quantidade; i++) {
                        Produtos p = new Produtos();

                        System.out.print("\nDigite o nome do produto: ");
                        p.setNome(read.next());

                        System.out.print("Digite a quantidade do produto: ");
                        p.setQuantidade(read.nextDouble());

                        System.out.print("Digite o preço do produto: ");
                        p.setPreco(read.nextDouble());

                        totalPedido = totalPedido + (p.getQuantidade() * p.getPreco());

                        linha = p.getNome() + "," + p.getQuantidade() + "," + p.getPreco() + "\n";
                        
                        try (BufferedWriter pw = new BufferedWriter(new FileWriter(path, true))) {
                            pw.write(nome + "\n");
                            pw.write(linha);
                            pw.newLine();
                        } 
                        
                        catch (IOException e) {
                            System.out.print("Erro ao escrever no arquivo");
                            break;
                        }
                    }

                    linha = nome + " -> R$ " + String.format("%.2f", totalPedido);
                    tw.write(linha);
                    tw.newLine();

                    System.out.print("Deseja continuar? (S/N)");
                    resposta = read.next().toUpperCase();
                    
                    if (resposta.equalsIgnoreCase("s")) {
                        continuar = true;
                    } 
                    
                    else {
                        continuar = false;
                    }
                } 
                
                catch (IOException e) {
                    System.out.print("Erro para ler o arquivo pedidos!");
                    break;
                }
            }
        } 
        
        catch (IOException e) {
            System.out.print("Erro ao criar o novo arquivo - total de pedidos!");
        }

    }
}

File: Python #5/Programa.py
def cadastro():
    nome = input("Digite seu nome: ")

    with open("pedidos.txt", "a") as arquivo:
        arquivo.write(f"{nome}\n")

        produto = int(input("Digite a quantidade de produtos que deseja comprar: "))

        for i in range(produto):
            produtos = input(f"Digite o nome do produto: ")
            quantidade = float(input(f"Digite a quantidade: "))
            preco = float(input(f"Digite o preço do produto: "))

            arquivo.write(f"{produtos}, {quantidade}, {preco}\n")

        arquivo.write("\n")

    continuar = input("Deseja continuar? (S/N) ")

    if continuar.upper() == "S":
        cadastro()

    else:
        calcularPedidos()

def calcularPedidos():

    with open("pedidos.txt", "r") as pedidos, open("total_pedidos.txt", "w") as totalA:

        totalCliente = {
            
        }

        for linha in pedidos:

            if linha.strip(): 

                if linha.endswith("\n"): 
                    linha = linha[:-1]

                if "," in linha: 
                    produtos, quantidade, preco = linha.split(",") 
                    total = float(quantidade) * float(preco)
                    totalCliente[nome].append(total)

                else:
                    nome = linha
                    totalCliente[nome] = []

        for nome, totalPedido in totalCliente.items():

            totalPedidos = sum(totalPedido)
            totalA.write(f"{nome.strip()} - R$ {totalPedidos:.2f}\n")

cadastro()